oci/private/image.sh

```diff
@@ -1,144 +1,174 @@
-            if [[ "${in}" = *\$* ]]; then
-              ENV_EXPANSIONS+=( "${in}" )
-            else
-              FIXED_ARGS+=( "--env=${in}" )
-            fi
-          done <"${ARG#--env-file=}"
-          ;;
-        (--labels-file=*)
-          while IFS= read -r in || [ -n "$in" ]; do
-            FIXED_ARGS+=("--label=$in")
-          done <"${ARG#--labels-file=}"
-          ;;
-        (--annotations-file=*)
-          while IFS= read -r in || [ -n "$in" ]; do
-            FIXED_ARGS+=("--annotation=$in")
-          done <"${ARG#--annotations-file=}"
-          ;;
-        (--cmd-file=*)
-          while IFS= read -r in || [ -n "$in" ]; do
-            FIXED_ARGS+=("--cmd=$in")
-          done <"${ARG#--cmd-file=}"
-          ;;
-        (--entrypoint-file=*)
-          while IFS= read -r in || [ -n "$in" ]; do
-            FIXED_ARGS+=("--entrypoint=$in")
-          done <"${ARG#--entrypoint-file=}"
-          ;;
-        (--exposed-ports-file=*)
-          while IFS= read -r in || [ -n "$in" ]; do
-                  FIXED_ARGS+=("--exposed-ports=$in")
-                done <"${ARG#--exposed-ports-file=}"
-                ;;
-              (*) FIXED_ARGS+=( "${ARG}" )
-          esac
+  case "$ARG" in
+  --scratch=*)
+    base_from_scratch "${ARG#--scratch=}"
+    ;;
+  --from=*)
+    base_from "${ARG#--from=}"
+    ;;
+  --layer=*) add_layer "${ARG#--layer=}" ;;
+  --env=*)
+    # Get environment from existing config
+    env=$(get_config | jq '(.config.Env // []) | map(. | split("=") | {"key": .[0], "value": .[1:] | join("=")})')
+
+    while IFS= read -r expansion || [ -n "$expansion" ]; do
+      IFS="=" read -r key value <<<"${expansion}"
+      value_from_base=$(jq -nr --arg raw "${value}" --argjson envs "${env}" "${ENV_EXPAND_FILTER}")
+      env=$(
+        # update the existing env if it exists, or append to the end of env array.
+        jq -r --arg key "$key" --arg value "$value_from_base" '. |= (map(.key) | index($key)) as $i | if $i then .[$i]["value"] = $value else . + [{key: $key, value: $value}] end' <<<"$env"
+      )
+    done <"${ARG#--env=}"
+
+    CONFIG=$(jq --argjson envs "${env}" '.config.Env = ($envs | map("\(.key)=\(.value)"))' <<<"$CONFIG")
+    ;;
+  --cmd=*)
+    CONFIG=$(jq --rawfile cmd "${ARG#--cmd=}" '.config.Cmd = ($cmd | split(",|\n"; "") | map(select(. | length > 0)))' <<<"$CONFIG")
+    ;;
+  --entrypoint=*)
+    CONFIG=$(jq --rawfile entrypoint "${ARG#--entrypoint=}" '.config.Entrypoint = ($entrypoint | split(",|\n"; "") | map(select(. | length > 0)))' <<<"$CONFIG")
+    ;;
+  --exposed-ports=*)
+    CONFIG=$(jq --rawfile ep "${ARG#--exposed-ports=}" '.config.ExposedPorts = ($ep | split(",") | map({key: ., value: {}}) | from_entries)' <<<"$CONFIG")
+    ;;
+  --user=*)
+    CONFIG=$(jq --arg user "${ARG#--user=}" '.config.User = $user' <<<"$CONFIG")
+    ;;
+  --workdir=*)
+    CONFIG=$(jq --arg workdir "${ARG#--workdir=}" '.config.WorkingDir = $workdir' <<<"$CONFIG")
+    ;;
+  --labels=*)
```

Comment:
if it's no longer a file, does that mean you can't `--stamp` values into them?

Reply:
All that said, i can take that back if you have a strong reason.

Reply:
nope that answers it, if they're still files then they can be stamped 👍🏻